### PR TITLE
Early cache enable and coherency fixes

### PR DESCRIPTION
**Pull request by glneo**

### CODE REVIEW

File: lib/psci/psci_common.c [hunk 1]
```diff
@@ -267,7 +267,7 @@ static plat_local_state_t *psci_get_req_local_pwr_states(unsigned int pwrlvl,
 static plat_local_state_t get_non_cpu_pd_node_local_state(
 		unsigned int parent_idx)
 {
-#if !USE_COHERENT_MEM || !HW_ASSISTED_COHERENCY
+#if !(USE_COHERENT_MEM || HW_ASSISTED_COHERENCY)
```
Comment (soby-mathew) [inline lib/psci/psci_common.c]: Yes, this is the right check.  This optimizes away the cache flush if either one of the build flag is enabled.

**Comment by soby-mathew** [comment]
IP-Review +1

**Comment by soby-mathew** [comment]
jenkins: test this please

**Comment by danh-arm** [comment]
jenkins: retest this please

**Pull request merged**